internal/observability/middleware_test.go

```diff
@@ -1371,6 +1371,110 @@ func TestMiddlewareSuccessSnapshot(t *testing.T) {
 	assert.Equal(t, want, snap, "Unexpected snapshot of metrics.")
 }
 
+func TestMiddlewareSuccessSnapshotWithTagsFiltered(t *testing.T) {
```

Comment:
nit: we should rename the test to `TestMiddlewareSuccessSnapshotWithTagsBlockList`